elements/lisk-api-client/src/codec.ts

```diff
@@ -40,21 +43,32 @@ export const getTransactionParamsSchema = (
 ): Schema | undefined => {
 	const moduleMeta = metadata.find(meta => meta.name === transaction.module);
 	if (!moduleMeta) {
-		throw new Error(
-			// eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-			`Module: ${transaction.module} is not registered.`,
-		);
+		throw new Error(`Module: ${transaction.module} is not registered.`);
 	}
 	const commandMeta = moduleMeta.commands.find(meta => meta.name === transaction.command);
 	if (!commandMeta) {
 		throw new Error(
-			// eslint-disable-next-line @typescript-eslint/restrict-template-expressions
 			`Module: ${transaction.module} CommandID: ${transaction.command} is not registered.`,
 		);
 	}
 	return commandMeta.params;
 };
 
+export const getEventDataSchema = (
+	event: { module: string; name: string },
+	metadata: ModuleMetadata[],
+): Schema | undefined => {
+	const moduleMetadata = metadata.find(meta => meta.name === event.module);
+	if (!moduleMetadata) {
+		throw new Error(`Module: ${event.module} is not registered.`);
+	}
+	const eventMetadata = moduleMetadata.events.find(meta => meta.name === event.name);
+	if (!eventMetadata) {
+		return standardEventDataSchema;
```

Comment:
`standardEventDataSchema` should be only used for `name == commandExecutionResult`